### PR TITLE
it have been added relationship for group, courses. Encoded password,…

### PR DESCRIPTION
**Pull request by yapollo2002**
… readById return List of UserDto.

### CODE REVIEW

File: src/main/java/ita/softserve/course_evaluation/dto/UserDtoMapper.java [hunk 1]
```diff
@@ -27,8 +35,9 @@ public static User fromDto(UserDto dto) {
 		user.setFirstName(dto.getFirstName());
 		user.setLastName(dto.getLastName());
 		user.setEmail(dto.getEmail());
-		user.setPassword(dto.getPassword());
+		user.setPassword(passwordEncoder.encode(dto.getPassword()));
```
Comment (Arsen-Kushnir) [inline src/main/java/ita/softserve/course_evaluation/dto/UserDtoMapper.java:38]: This was already happening in the AuthController during registration

File: src/main/java/ita/softserve/course_evaluation/controller/UserController.java [hunk 2]
```diff
@@ -39,12 +39,12 @@ public UserController(UserService userService) {
 
 	@GetMapping("/{id}")
 	public ResponseEntity<UserDto> readById(@PathVariable(value="id") long id){
-		return new ResponseEntity<>(userService.readById(id),  HttpStatus.OK);
+		return new ResponseEntity                                                                                                                                                                                   <>(userService.readById(id),  HttpStatus.OK);
```
Comment (Arsen-Kushnir) [inline src/main/java/ita/softserve/course_evaluation/controller/UserController.java:42]: Empty ResponseEntity?

**Review by Arsen-Kushnir — CHANGES_REQUESTED** [review]

File: src/main/java/ita/softserve/course_evaluation/dto/UserDtoMapper.java (hunk 1 above)
Comment (yapollo2002) [reply src/main/java/ita/softserve/course_evaluation/dto/UserDtoMapper.java:40]: it is for Admin mode meybe usefull

File: src/main/java/ita/softserve/course_evaluation/controller/UserController.java (hunk 2 above)
Comment (yapollo2002) [reply src/main/java/ita/softserve/course_evaluation/controller/UserController.java]: one part of code has been shifed to the right, corrected

**Review by Arsen-Kushnir — APPROVED** [review]

**Pull request merged**